SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -2691,6 +2796,114 @@ void USpatialNetDriver::DelayedRetireEntity(Worker_EntityId EntityId, float Dela
 		Delay, false);
 }
 
+void USpatialNetDriver::QueryRoutingPartition()
+{
+	if (bRoutingWorkerQueryInFlight)
+	{
+		// Only allow one in flight query. Retries will be handled by the SpatialNetDriver.
+		return;
+	}
+
+	if (RoutingWorkerId == 0)
+	{
+		Worker_ComponentConstraint ComponentConstraint{};
+		ComponentConstraint.component_id = SpatialConstants::WORKER_COMPONENT_ID;
+
+		Worker_Constraint Constraint{};
+		Constraint.constraint_type = WORKER_CONSTRAINT_TYPE_COMPONENT;
+		Constraint.constraint.component_constraint = ComponentConstraint;
+
+		Worker_EntityQuery Query{};
+		Query.constraint = Constraint;
+
+		Worker_RequestId RequestID = Connection->SendEntityQueryRequest(&Query, SpatialGDK::RETRY_UNTIL_COMPLETE);
+		bRoutingWorkerQueryInFlight = true;
+
+		TWeakObjectPtr<USpatialNetDriver> WeakNetDriver(this);
+		EntityQueryDelegate RoutingWorkerQueryDelegate;
+		RoutingWorkerQueryDelegate.BindLambda([WeakNetDriver](const Worker_EntityQueryResponseOp& Op) {
+			if (!WeakNetDriver.IsValid())
+			{
+				return;
+			}
+
+			USpatialNetDriver* NetDriver = WeakNetDriver.Get();
+			if (Op.status_code == WORKER_STATUS_CODE_SUCCESS)
+			{
+				for (uint32_t i = 0; i < Op.result_count; ++i)
+				{
+					const Worker_Entity& Entity = Op.results[i];
+					for (uint32_t j = 0; j < Entity.component_count; ++j)
+					{
+						const Worker_ComponentData& ComponentData = Entity.components[j];
+						if (ComponentData.component_id == SpatialConstants::WORKER_COMPONENT_ID)
+						{
+							Schema_Object* Fields = Schema_GetComponentDataFields(ComponentData.schema_type);
+							FString WorkerId = SpatialGDK::GetStringFromSchema(Fields, 1);
+							FString WorkerType = SpatialGDK::GetStringFromSchema(Fields, 2);
+							if (WorkerType == SpatialConstants::RoutingWorkerType.ToString())
+							{
+								NetDriver->RoutingWorkerId = Entity.entity_id;
```

Comment:
Is ID=0 a valid entity ID? Could end up recursively in here if so

Reply:
I'm pretty sure it is not.